backend.native/tests/interop/concurrentTerminate/async.cpp

```diff
@@ -0,0 +1,25 @@
+#include <thread>
+#include <future>
+#include <chrono>
+#include <vector>
+#include <csignal>  // signal.h
+
+#include "async.h"
+
+int test_ConcurrentTerminate() {
+    signal(SIGABRT, *[](int){ exit(0); }); // Windows does not have sigaction
+
+    std::vector<std::future<void>> futures;
+
+    for (size_t i = 0; i < 100; ++i) {
+        futures.emplace_back(std::async(std::launch::async,
+                [](size_t param) {
+                    std::this_thread::sleep_for(std::chrono::milliseconds(param));
```

Comment:
Just for the record (i.e. not doubting your approach), I used the following pattern to induce races:
```cpp
int launchedThreads = 0;
bool threadsCanContinue = false;
for (int i = 0; i < numberOfThreads; ++i) {
  startAThread([]() {
    launchedThreads += 1; // increment atomically
    while (!threadsCanContinue) {} // read atomically
    // Do a racy thing here.
  });
}
while (launchedThreads < numberOfThreads) {} // read atomically
threadsCanContinue = true;  // write atomically
```
On my machine this pattern turned out to be quite reliable.

Reply:
Thanks! Indeed`future` seems to be a bit "indirect" way, in comparison with `startAtThread`.
My snippet is derived from more complicate test involving exception propagation with `promise` and `set_exception`.

Reply:
I think `std::async` would work just fine either way. My point was in using spinlocks for synchronisation as opposed to sleeping.